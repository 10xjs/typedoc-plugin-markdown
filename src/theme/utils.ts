import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
