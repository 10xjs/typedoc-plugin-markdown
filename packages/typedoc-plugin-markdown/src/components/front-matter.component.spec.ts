import { PageEvent } from 'typedoc/dist/lib/output/events';

import { TestApp } from '../../test/utils';
import { FrontMatterComponent } from './front-matter.component';

describe(`FrontMatterComponent`, () => {
  let testApp: TestApp;
  let frontMatterComponent: FrontMatterComponent;
  beforeAll(() => {
    testApp = new TestApp(['variables']);
    testApp.bootstrap();
    testApp.renderer.addComponent(
      'frontmatter',
      new FrontMatterComponent(testApp.renderer),
    );
    frontMatterComponent = testApp.renderer.getComponent(
      'frontmatter',
    ) as FrontMatterComponent;
  });

  test(`should prepend YAML block to start of page`, () => {
    const page = {
      contents: '[CONTENT]',
      url: 'modules/_access_access_.md',
      model: { name: '"access/access"' },
      project: { url: 'index.md' },
    } as PageEvent;
    frontMatterComponent.onPageEnd(page);
    expect(page.contents).toMatchSnapshot();
  });
});
