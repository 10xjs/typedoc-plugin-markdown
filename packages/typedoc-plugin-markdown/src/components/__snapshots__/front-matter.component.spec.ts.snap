// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontMatterComponent should prepend YAML block to start of page 1`] = `
"---
id: \\"_access_access_\\"
title: \\"access/access\\"
---

[CONTENT]"
`;
